daphne/src/messages/mod.rs

```diff
@@ -1130,34 +1114,44 @@ impl Decode for HpkeCiphertext {
 #[derive(Clone, Debug, PartialEq, Eq)]
 #[allow(missing_docs)]
 pub struct PlaintextInputShare {
-    pub extensions: Vec<Extension>,
+    pub draft07_extensions: Option<Vec<Extension>>,
     pub payload: Vec<u8>,
 }
 
-impl Encode for PlaintextInputShare {
-    fn encode(&self, bytes: &mut Vec<u8>) {
-        encode_u16_items(bytes, &(), &self.extensions);
-        encode_u32_bytes(bytes, &self.payload);
+impl ParameterizedEncode<DapVersion> for PlaintextInputShare {
+    fn encode_with_param(&self, version: &DapVersion, bytes: &mut Vec<u8>) {
+        match (version, &self.draft07_extensions) {
+            (DapVersion::Draft07, Some(extensions)) => {
+                encode_u16_items(bytes, &(), extensions);
+                encode_u32_bytes(bytes, &self.payload);
+            }
+            (DapVersion::Draft07, None) => {
+                unreachable!("PlaintextInputShare with missing extensions")
+            }
+            (DapVersion::Draft02, ..) => {
+                unreachable!("cannot encode PlaintextInputShare in draft02")
+            }
+        }
```

Comment:
This feels wrong:
- you're taking a version parameter but it's never possible to use any variant other than `Draft07`, which means you're passing more information but only use it to panic
- the type allows for `None` extensions but this is an unencodable state, which leads me to think it's also an invalid state for the type

Reply:
Ugh, yeah you're absolutely right. I removed the decoding parameter and cleaned up, please take another look.